### PR TITLE
Make .json easier to use and include more data

### PR DESCRIPTION
**Pull request by asdil12**
The current json output is painful to parse as it violates the hierarchical principle of json.

- Don't keep data structures of different types in the same hash (move `_group` and `_max` one level up)
- Expose comments via job_group .json

### CODE REVIEW

**Comment by okurz** [comment]
It might be surprising but the name is actually "painful" not painfull :-)

**Comment by okurz** [comment]
seems ok to me

**Comment by coolo** [comment]
damn, this is introducing an old regression again - we failed to add a test. The dashboard shows *all* job groups again, not only those with builds

File: lib/OpenQA/BuildResults.pm [hunk 1]
```diff
@@ -110,15 +110,13 @@ sub compute_build_results {
         $builds{$b}              = \%jr;
         $max_jobs = $count if ($count > $max_jobs);
     }
-    if (%builds) {
-        $builds{_max}   = $max_jobs;
-        $builds{_group} = {
+    return {
+        result => (%builds) ? \%builds : {},
```
Comment (coolo) [inline lib/OpenQA/BuildResults.pm]: the if had a different purpose. But please add a proper test case

**Pull request merged**